kernel: Fix readid method and oob layout of XTX XT26G0xA.

0xFelix opened the pull request:
Fix readid method and oob layout of XTX XT26G0xA.

This fixes https://github.com/openwrt/openwrt/issues/5091 in `master`.

981213 commented:
> The correct readid method is `SPINAND_READID_METHOD_OPCODE_DUMMY`.

Typo :) It's SPINAND_READID_METHOD_OPCODE_ADDR

0xFelix commented:
Oops 😄 fixed that

0xFelix commented:
Please put this on hold until. The OOB layout also needs to be corrected.

0xFelix committed:
kernel: Fix oob layout of XTX XT26G0xA

The correct oob layout is:

ECC:
region->offset = 48;
region->length = 16;

Free:
/* Reserve 1 byte for the BBM. */
region->offset = 1;
region->length = 47;

Signed-off-by: Felix Matouschek <felix@matouschek.org>

0xFelix committed:
kernel: Fix readid method of XTX XT26G0xA

The correct readid method is SPINAND_READID_METHOD_OPCODE_ADDR.

Signed-off-by: Felix Matouschek <felix@matouschek.org>

0xFelix commented:
The OOB layout is now also fixed. It can be merged now.

981213 commented:
Merged. Thanks!
